### PR TITLE
Delay initialization of the OC\Encryption\Update - introducing 'OC\Encry...

### LINKED ISSUE #15693: [enc2] Unrecoverable state if encryption app triggers NeedsUpdateException in OC_App::loadApp()
```json
{"reqId":"oJB+a0ZowwsrRhcVMJX3",
"remoteAddr":"178.25.224.1",
"app":"index",
"message":"Exception: {\"Message\":\"\",\"Code\":0,\"Trace\":\"#0 \\\/srv\\\/www\\\/owncloud8.1\\\/lib\\\/private\\\/app.php(99): OC_App::loadApp('encryption')\\n#1 \\\/srv\\\/www\\\/owncloud8.1\\\/lib\\\/private\\\/util.php(127): OC_App::loadApps(Array)\\n#2 \\\/srv\\\/www\\\/owncloud8.1\\\/lib\\\/private\\\/files\\\/filesystem.php(235): OC_Util::setupFS()\\n#3 \\\/srv\\\/www\\\/owncloud8.1\\\/lib\\\/base.php(733): OC\\\\Files\\\\Filesystem::getMountManager()\\n#4 \\\/srv\\\/www\\\/owncloud8.1\\\/lib\\\/base.php(657): OC::registerEncryptionHooks()\\n#5 \\\/srv\\\/www\\\/owncloud8.1\\\/lib\\\/base.php(1118): OC::init()\\n#6 \\\/srv\\\/www\\\/owncloud8.1\\\/index.php(38): require_once('\\\/srv\\\/www\\\/ownclo...')\\n#7 {main}\",\"File\":\"\\\/srv\\\/www\\\/owncloud8.1\\\/lib\\\/private\\\/app.php\",\"Line\":118}",
"level":4,
"time":"2015-04-17T11:16:32+00:00"}

### PR DESCRIPTION
**Pull request by DeepDiver1975**
...ption\HookManager'

fixes #15693

@schiesbn @th3fallen @PVince81 THX

How to reproduce the issue:
- enable encryption app
- enable encryption
- change version of encryption app
- reload

-> :boom:

### CODE REVIEW

**Comment by PVince81** [comment]
Tested before and after this fix and I can confirm that the update worked :+1:

**Comment by karlitschek** [comment]
:+1:

File: lib/private/encryption/hookmanager.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+		self::getUpdate()->postShared($params);
+	}
+	public static function postUnshared($params) {
+		self::getUpdate()->postUnshared($params);
+	}
+
+	/**
+	 * @return Update
+	 */
+	private static function getUpdate() {
```
Comment (th3fallen) [inline lib/private/encryption/hookmanager.php]: Kinda late now but why static?

Comment (DeepDiver1975) [inline lib/private/encryption/hookmanager.php]: because everything in that class is static

**Pull request merged**